### DISCUSSION TITLE
RoleToPermission and Permission entity in EF data model

### DISCUSSION THREAD

**Discussion by mehdihadeli** [kgrzybek/modular-monolith-with-ddd#195] [Ideas] [answered]
Hi @kgrzybek,
I can't find implementation for `RoleToPermission` and `Permission` in the domain model of your code base, but their tables exists in your sql migration scripts!! 
How could we map `RoleToPermission` configurations for EF core?

**Comment by AndreiGanichev** [comment]
Hi, @mehdihadeli.
There is `v_UserPermissions` [view](https://github.com/kgrzybek/modular-monolith-with-ddd/blob/37de06228f3f6f49de20a510d5349c784903b01f/src/Database/CompanyName.MyMeetings.Database/Scripts/Migrations/1_0_0_0/0001_initial_structure.sql#L890) which contains data about users and their permissions. This view is used in **Read model** to [get](https://github.com/kgrzybek/modular-monolith-with-ddd/blob/master/src/Modules/UserAccess/Application/Authorization/GetAuthenticatedUserPermissions/GetAuthenticatedUserPermissionsQueryHandler.cs) the user's permissions and [check](https://github.com/kgrzybek/modular-monolith-with-ddd/blob/master/src/API/CompanyName.MyMeetings.API/Configuration/Authorization/HasPermissionAuthorizationHandler.cs) is the user authorized for some action. Read model doesn't use EF(Write model does).

btw, @kgrzybek wrote about using database views [here](http://www.kamilgrzybek.com/design/simple-cqrs-implementation-with-raw-sql-and-ddd/):
> I try not to refer to tables directly and instead refer to database views. This is a nice way to create abstraction and decouple our application from database schema because I want to hide database internals as much as possible.

**Comment by mehdihadeli** [comment, marked as answer]
@AndreiGanichev Thanks for your explanation.
